cmd/cli_test.go

```diff
@@ -345,6 +334,81 @@ func run(t *testing.T, test testCase) {
 	}
 }
 
+// Simple explanation:
+// basically we leverage text.template to execute templates for test data.
+// testTemplateData is the actual template structure used to fill the test requested output.
+// It internally stores all the help text, split in section { Desc, Usage, Commands, Flags, Info }
+// plus the flagsTemplateData (that is the template data used to fill Flags fields basically).
+type testTemplateData struct {
+	Desc     string
+	Usage    string
+	Commands string
+	Flags    string
```

Comment:
Note that flags is itself a templated script ;) therefore we need both the
* `testTemplateData` -> to be passed to actual test template
* `flagsTemplateData` -> to be used by testTemplateData to compute its `Flags` field.